mlir/test/lit.cfg.py

```diff
@@ -168,6 +168,7 @@ def add_runtime(name):
         ToolSubst("transform-opt-ch3", unresolved="ignore"),
         ToolSubst("transform-opt-ch4", unresolved="ignore"),
         ToolSubst("mlir-transform-opt", unresolved="ignore"),
+        ToolSubst("%llvm_shlib_dir", config.llvm_shlib_dir, unresolved="ignore"),
```

Comment:
but these aren't llvm shlibs? they're mlir shlibs? for which there's a substitution right below? are you, in downstream, setting `config.llvm_shlib_dir` to something and then (currently) it's not getting picked up?

Reply:
I'm indeed working downstream on Fedora.
The problem is that we built MLIR standalone. That means `../../../../` goes to an unrelated directory.
I agree with you that, while it works, `llvm_shlibs_dir` is not the ideal solution.

`mlir_lib_dir` also does not point to the right place where these libraries are available at build time.

Looking at other tests, I noticed they use `%mlir_runner_utils` and `%mlir_c_runner_utils` in order to find these libraries. I'm changing this PR in order to use them.